tests/zkevm/test_worst_compute.py

```diff
@@ -104,3 +117,207 @@ def test_worst_keccak(
+            
+            # Align the parentheses by padding the values section
+            f.write(f"{description:<30} {bytecode_values:>15} ({code_size_pct:6.2f}%)  {gas_values:>20} ({gas_pct:6.2f}%)  {bytes_per_iter:>4}  {num_iters:>10}\n")
+            
+        LOGGED_DESCRIPTIONS.add(description)
+    
+    if len(code) > MAX_CODE_SIZE:
+        # Must never happen, but keep it as a sanity check.
+        raise ValueError(f"Code size {len(code)} exceeds maximum code size {MAX_CODE_SIZE}")
+
+    code_address = pre.deploy_contract(code=code)
+
+    tx = Transaction(
+        to=code_address,
+        gas_limit=gas_limit,
+        gas_price=10,
+        sender=pre.fund_eoa(),
+        data=[],
+        value=0,
+    )
+
+    return blockchain_test(
+        env=env,
+        pre=pre,
+        post={},
+        blocks=[Block(txs=[tx])],
+    )
+
+
+@pytest.mark.zkevm
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.parametrize(
+    "gas_limit",
+    [
+        36_000_000,
+    ],
+)
+def test_jumpdest_only(
+    blockchain_test: BlockchainTestFiller,
+    pre: Alloc,
+    fork: Fork,
+    gas_limit: int,
+):
+    """Test with only JUMPDEST opcodes"""
+    gsc = fork.gas_costs()
+    iteration_gas_cost = gsc.G_JUMPDEST
+
+    # Use Op.JUMPDEST instead of raw bytes
+    opcodes_pattern = Op.JUMPDEST
+
+    run_jump_analysis_test(
+        blockchain_test=blockchain_test,
+        pre=pre,
+        fork=fork,
+        gas_limit=gas_limit,
+        opcodes_pattern=opcodes_pattern,
+        iteration_gas_cost=iteration_gas_cost,
+        description="JUMPDEST only",
+    )
+
+
+@pytest.mark.zkevm
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.parametrize(
+    "gas_limit",
+    [
+        36_000_000,
+    ],
+)
+def test_jumpdest_jump(
+    blockchain_test: BlockchainTestFiller,
+    pre: Alloc,
+    fork: Fork,
+    gas_limit: int,
+):
+    """Test with JUMPDEST and JUMP opcodes"""
+    gsc = fork.gas_costs()
+    iteration_gas_cost = gsc.G_JUMPDEST + gsc.G_MID
+
+    # Use Op.JUMPDEST + Op.JUMP instead of raw bytes
+    opcodes_pattern = Op.JUMPDEST + Op.JUMP
+
+    run_jump_analysis_test(
+        blockchain_test=blockchain_test,
+        pre=pre,
+        fork=fork,
+        gas_limit=gas_limit,
+        opcodes_pattern=opcodes_pattern,
+        iteration_gas_cost=iteration_gas_cost,
+        description="JUMPDEST + JUMP",
+    )
+
+
+@pytest.mark.zkevm
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.parametrize(
+    "gas_limit",
+    [
+        36_000_000,
+    ],
+)
+def test_jumpdest_push0_pop_jump(
+    blockchain_test: BlockchainTestFiller,
+    pre: Alloc,
+    fork: Fork,
+    gas_limit: int,
+):
+    """Test with JUMPDEST, PUSH0, POP, and JUMP opcodes"""
+    gsc = fork.gas_costs()
+    iteration_gas_cost = gsc.G_JUMPDEST + gsc.G_VERY_LOW + gsc.G_BASE + gsc.G_MID
+
+    # Use Op.JUMPDEST + Op.PUSH0 + Op.POP + Op.JUMP instead of raw bytes
+    opcodes_pattern = Op.JUMPDEST + Op.PUSH0 + Op.POP + Op.JUMP
```

Comment:
What is the theory behind this setup? Do you think that mixing some PUSH0/POP with JUMPs can create some particular interesting case?

Reply:
No this can go away